network/betweenness.go

```diff
@@ -32,9 +32,61 @@ func Betweenness(g graph.Graph) map[int]float64 {
 	// Also note special case for sparse networks:
 	// http://wwwold.iit.cnr.it/staff/marco.pellegrini/papiri/asonam-final.pdf
 
-	var (
-		cb = make(map[int]float64)
+	cb := make(map[int]float64)
+	brandes(g, func(s graph.Node, stack internal.NodeStack, p map[int][]graph.Node, delta, sigma map[int]float64) {
+		for stack.Len() != 0 {
+			w := stack.Pop()
+			for _, v := range p[w.ID()] {
+				delta[v.ID()] += sigma[v.ID()] / sigma[w.ID()] * (1 + delta[w.ID()])
+			}
+			if w.ID() != s.ID() {
+				if d := delta[w.ID()]; d != 0 {
+					cb[w.ID()] += d
+				}
+			}
+		}
+	})
+	return cb
+}
+
+// EdgeBetweenness returns the non-zero betweenness centrality for edges in the unweighted graph g.
+//
+//  C_B(q(u, v)) = \sum_{s ≠ t, u ≠ v ∈ V} (\sigma_{st}(q) / \sigma_{st})
```

Comment:
I'll go with e, then since I'm not keeping nodes (this explains why I was confused with the last conditional in 7 - his C_B[q] give the node betweenness as well - odd thing to do when you're trying to explain derivations). Thanks for clearing up that quandary.